docs/general/dql/selects.rst

```diff
@@ -740,6 +740,34 @@ an upper-bound::
     Both the ``from`` and ``to`` index values are inclusive.
     Using an index greater than the array size results in an empty array.
 
+For selecting records having values with empty or unassigned arrays, like::
+
+    cr> insert into locations (id, name, position, kind, landmarks) values
+    ... (99, 'Voidspace', 0, 'Milliways', []);
+    INSERT OK, 1 row affected (... sec)
+
+.. Hidden: refresh locations
+
+    cr> refresh table locations;
+    REFRESH OK, 1 row affected (... sec)
+
+a suitable expression to match both conditions, is::
+
+    cr> select count(*) from locations where
+    ... landmarks is null or landmarks = [];
```

Comment:
I still don't really understand what's special about this construct that we feel the need to point this out specifically?

The other examples in the text educate the user about some specific behavior, but this here doesn't?

Reply:
Hi Mathias,

the snippet originated from my explorations around that topic, based on your response at https://github.com/crate/crate/issues/9398#issuecomment-1169750540, which I was not able to reproduce, because the `array_length()` of an empty array would not evaluate to zero, but `NULL` instead. This was good to learn.

As I recognized that there has been some confusion around using `array_length` for that purpose (https://github.com/crate/crate/issues/9398#issuecomment-559748613), my feeling was that it would be a good idea to add an explicit example now, specifically after this expression has been optimized with #12696.

With kind regards,
Andreas.